test-result-summary-client/src/Build/Summary/ResultSummary.jsx

```diff
@@ -155,6 +156,16 @@ export default class ResultSummary extends Component {
             } else {
                 console.warn(`Cannot match ${buildName}`);
             }
+
+            if (build.status != 'Done') {
+                childBuildsResult = 'PROGRESSING';
+            } else if (childBuildsResult == '') {
+                if (build.buildResult == 'FAILIURE') {
+                    childBuildsResult = 'FAILIURE';
+                } else if (build.buildResult == 'UNSTABLE') {
+                    childBuildsResult = 'UNSTABLE';
+                }
```

Comment:
In the above logic, we override childBuildsResult. The value may vary based on the loop. We can use the following logic:
```
let childBuildsResult = 'SUCCESS';
cont buildResultPriority = {ABORT: 3, FAILIURE:  2, UNSTABLE: 1, SUCCESS: 0};

....
if (build.status != 'Done') {
    childBuildsResult = 'PROGRESSING';
} else {
    childBuildsResult = Math.max(buildResultPriority[build.buildResult], childBuildsResult);
}
...
```